compiler/crates/resolution-path/src/test.rs

```diff
@@ -390,3 +413,1183 @@ fn fragment_argument_directive() {
         assert_matches!(resolved, ResolutionPath::ConstantString(_));
     })
 }
+
+// ## Directives
+
+#[test]
+fn directive_definition_name() {
+    let source = r#"
+        directive @foo on FIELD
+        "#;
+    test_schema_resolution(source, "foo", |resolved| {
+        assert_matches!(
+            resolved,
+            ResolutionPath::Ident(IdentPath {
+                inner: _,
+                parent: IdentParent::DirectiveDefinitionName(_),
+            })
+        );
+    })
+}
+
+#[test]
+fn directive_definition_argument_name() {
+    let source = r#"
+        directive @foo(bar: Baz) on FIELD
+        "#;
+    test_schema_resolution(source, "bar", |resolved| {
+        assert_matches!(
+            resolved,
+            ResolutionPath::Ident(IdentPath {
+                inner: _,
+                parent: IdentParent::InputValueDefinitionName(InputValueDefinitionPath {
+                    inner: _,
+                    parent: InputValueDefinitionParent::DirectiveDefinition(_),
+                }),
+            })
+        );
+    })
+}
+
+// ## Union Types
+
+#[test]
+fn union_type_definition_name() {
+    let source = r#"
+        union Foo = Bar | Baz
+        "#;
+    test_schema_resolution(source, "Foo", |resolved| {
+        assert_matches!(
+            resolved,
+            ResolutionPath::Ident(IdentPath {
+                inner: _,
+                parent: IdentParent::UnionTypeDefinitionName(_),
+            })
+        );
+    })
+}
+
+#[test]
+fn union_type_definition_member_name() {
+    let source = r#"
+        union Foo = Bar | Baz
+        "#;
+    test_schema_resolution(source, "Bar", |resolved| {
+        assert_matches!(
+            resolved,
+            ResolutionPath::Ident(IdentPath {
+                inner: _,
+                parent: IdentParent::UnionTypeMemberType(
+                    UnionTypeMemberParent::UnionTypeDefinition(_),
```

Comment:
What would be a good way to go about this? A union member type is just an Identifier, but it can have two parents (union definition and union extension). Would two IdentParents be better here?

Reply:
I think this is fine. This part of the code base is just super verbose. I'm not too worried about it.

That said, as you can tell, this ResolutionPath approach, while comprehensive, is pretty tedious to implement in Rust. I wonder what Rust Analyzer does. Maybe there's a simpler way to do this type of location lookup.